src/Core/SettingsQuirks.cpp

```diff
@@ -14,19 +15,22 @@
 ///   [2]: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=0c54a6a44bf3
 bool nestedEpollWorks(Poco::Logger * log)
 {
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 5, 0)) && (LINUX_VERSION_CODE < KERNEL_VERSION(5, 6, 13))
-        /// the check is correct since there will be no more 5.5.x releases.
+    if (Poco::Environment::os() != POCO_OS_LINUX)
+        return true;
+
+    DB::VersionNumber linux_version = DB::VersionNumber::fromString(
+        Poco::Environment::osVersion(), /* strict= */ false);
+
+    /// the check is correct since there will be no more 5.5.x releases.
+    if (linux_version >= DB::VersionNumber{5, 5, 0} && linux_version < DB::VersionNumber{5, 6, 13})
```

Comment:
Maybe simply refuse to start on these kernel versions instead of making quirks?

Reply:
I don't think that this is a good idea, since the benefit of async_socket_for_remote is doubtful to me (it has lots of syscall overhead at least), and use_hedged_requests is significant only for big cluster setup.